phac-nml/irida-next
# Update: Viral Button Component

## What does this PR do and why?

Designed a Viral Components button component to allow for consistently styled buttons in the UI.  Allows for passing html attributes to facilitate adding stimulus controller functionality and custom classes.

## Screenshots or screen recordings

### Default
![image](https://github.com/phac-nml/irida-next/assets/11295750/0bdea294-6ad3-4c87-85bd-4bae1fa99b82)

### Destructive
![image](https://github.com/phac-nml/irida-next/assets/11295750/a2b28ed6-4035-42bc-8312-887e41bd7eef)

### Disclosure
![image](https://github.com/phac-nml/irida-next/assets/11295750/85cfef36-1510-4898-8f4c-ccf0037171a3)

### Full Width (Set within a 300px div)
![image](https://github.com/phac-nml/irida-next/assets/11295750/d0f6c302-a383-4693-ab35-c3f5b61600bf)

### Large
![image](https://github.com/phac-nml/irida-next/assets/11295750/5c39ad9a-34da-4199-be47-3ed33226f28d)

### Primary
![image](https://github.com/phac-nml/irida-next/assets/11295750/517bf55a-1b79-49bc-a2b1-b99e7507dd10)

### Small
![image](https://github.com/phac-nml/irida-next/assets/11295750/639909be-9cf3-4915-a6b4-60e6dd94bd75)

## How to set up and validate locally

Running development server, go to the following LookBook previews, there you can see the required markup and HTML output:

1. [Basic](http://localhost:3000/rails/lookbook/inspect/viral_button/basic)
2. [Destructive](http://localhost:3000/rails/lookbook/inspect/viral_button/destructive)
3. [Disclosure](http://localhost:3000/rails/lookbook/inspect/viral_button/disclosure)
4. [Full Width](http://localhost:3000/rails/lookbook/inspect/viral_button/full_width)
5. [Large](http://localhost:3000/rails/lookbook/inspect/viral_button/large)
6. [Primary](http://localhost:3000/rails/lookbook/inspect/viral_button/primary)
7. [Small](http://localhost:3000/rails/lookbook/inspect/viral_button/small)
